### DISCUSSION TITLE
Add viewport meta tag

### DISCUSSION THREAD

**Discussion by KaKi87** [openstyles/stylus#1619] [Ideas]
Hello,

When making a userstyle that adds responsive design to a desktop-only website, JS is required to add the [viewport meta tag](https://developer.mozilla.org/en-US/docs/Web/HTML/Viewport_meta_tag) :

```js
const metaViewportElement = document.createElement('meta');
metaViewportElement.setAttribute('name', 'viewport');
metaViewportElement.setAttribute('content', 'width=device-width, initial-scale=1');
document.head.appendChild(metaViewportElement);
```

Therefore requiring a userscript.

It would be nice for Stylus to be able to add that tag, e.g. using a UserCSS metadata like the following :

```
@viewport width=device-width, initial-scale=1
```

In which case such a userstyle wouldn't need to be distributed as a userscript.

Thanks

**Comment by saxamaphone69** [comment]
I'm curious, why do you need to add that HTML to make it responsive? `viewport` is used, more or less, to make websites viewed on a mobile/tablet, "zoom" in. AFAIK, it doesn't matter to websites viewed on a desktop. What are you trying that isn't achievable with media queries?

**Reply by KaKi87** [reply]
Here's a userscript I made for `old.reddit.com` :

![](https://github.com/openstyles/stylus/assets/21284089/8e412037-d46d-441d-aa3c-b6946520d9a5)

Here's what happens if I comment the code that adds the viewport meta tag :

![](https://github.com/openstyles/stylus/assets/21284089/a1e370d2-d5dc-4a78-86fb-e451abf9576d)

Which is normal, because that tag is necessary.

**Reply by KaKi87** [reply]
Hello,
Any news on this ?
Thanks

**Reply by tophf** [reply]
I don't think we should fix sites...

**Reply by KaKi87** [reply]
Isn't userstyles all about fixing bad UI ?

**Reply by tophf** [reply]
Stylus only runs userstyles, it doesn't fix sites.

**Reply by krystian3w** [reply]
Check these if you allow use Tampermonkey:

https://greasyfork.org/scripts/391314-auto-meta-viewport

My hotfix and limit to `old.reddit.com` only:
```js
// ==UserScript==
// @name           Auto meta viewport
// @namespace      https://github.com/NateScarlet/Scripts/tree/master/user-script
// @description    Use <meta name="viewport" content="width=device-width, initial-scale=1"> as default.
// @version        2222.01
// @run-at         document-end
// @match          old.reddit.com
// ==/UserScript==
function main() {
    if (document.querySelector('meta[name="viewport"]')) {
        return;
    }
    const el = document.createElement("meta");
    el.name = "viewport";
    el.content = "width=device-width, initial-scale=1";
    document.head.append(el);
}
main();
```

**Reply by KaKi87** [reply]
That's what I did, if you'd bother to read my original post.

**Comment by krystian3w** [comment]
Years ago:
- 2015: http://web.archive.org/web/20150819085732/https://drafts.csswg.org/css-device-adapt/#atviewport-rule
- 2020 - 28 August 2022: https://web.archive.org/web/20200508225021/https://drafts.csswg.org/css-device-adapt/#atviewport-rule
